### PR TITLE
Clojure FDK

### PR DESCRIPTION
**Pull request by jgh-**
Hello,  I have created a Clojure FDK from code I've been working on for my project.  The FDK itself can be found at https://github.com/unpause-live/fdk-clj.  It currently supports CloudEvents and JSON format and passes Fn Test.

In this PR is a Clojure language helper which will, as with the other languages, prepare some example boilerplate.   The supplied tests in the boilerplate pass as well.

In the current example I have it building with `clojure:lein` and running with `openjdk:9-jre`, however functions could also be built using Graal's `native-image` and run from `SCRATCH`.

### CODE REVIEW

File: langs/clojure.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        fullPath := filepath.Join(path, dir)
+        if err = os.MkdirAll(fullPath, os.FileMode(0755)); err != nil {
+            return err
+        }
+
+        fullFilePath := filepath.Join(fullPath, filename)
+        return ioutil.WriteFile(fullFilePath, []byte(content), os.FileMode(0644))
+    }
+
+    err = mkDirAndWriteFile("src/example", "core.clj", helloClojureSrcBoilerplate)
```
Comment (denismakogon) [inline langs/clojure.go]: let's call it `src/func`, we `func` across multiple lang helpers, so, let's stick with that

File: langs/base.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ func init() {
 	registerHelper(&RubyLangHelper{})
 	registerHelper(&RustLangHelper{})
 	registerHelper(&KotlinLangHelper{})
+	registerHelper(&ClojureLangHelper{version: "1.9.0"})
```
Comment (denismakogon) [inline langs/base.go]: overall, looks good, need to do few manual tests.

in a meantime, can you please add tests for this helper?
take a look at:
 - https://github.com/fnproject/cli/blob/master/test/cli_lang_boilerplate_test.go#L15

Comment (jgh-) [reply langs/base.go]: Sure, will do.

**Comment by denismakogon** [comment]
Would be nice to write an article on medium about that.

**Comment by jgh-** [comment]
OK, I'm not much of a blogger but happy to write an article about it!  It's something I should be doing more of anyway.

**Comment by mjg123** [comment]
Yes - agree.  I'm aiming to get the init-image stuff at #369 merged asap so that this can be supported.  I do ❤️ clojure after all 😄

**Comment by jgh-** [comment]
OK, I will revisit this when #369 is merged and figure out the correct way for the Clojure FDK to be compliant!

**Comment by mjg123** [comment]
We're documenting it now, but roughly what we will need is a docker image which will be run with something like:  `docker run -e FN_FUNCTION_NAME=<the_name> <your_image>`.  The output should be something which can be piped to `tar -x` to produce the files on disk.  We will be creating several examples which you can copy from.

**Pull request closed without merging**